server/utils/appInsightsUtils.ts

```diff
@@ -1,13 +1,43 @@
 import type { TelemetryClient } from 'applicationinsights'
-import { DistributedTracingModes, defaultClient, setup } from 'applicationinsights'
+import { Contracts, DistributedTracingModes, defaultClient, setup } from 'applicationinsights'
+import type { EnvelopeTelemetry } from 'applicationinsights/out/Declarations/Contracts'
 
 import { buildNumber, packageData } from '../applicationVersion'
+import type { User } from '../data'
+
+type ContextObjectWithUser = {
+  res?: {
+    locals?: {
+      user: Partial<User>
+    }
+  }
+}
+
+export type ContextObject = Record<string, ContextObjectWithUser> | undefined
 
 export default class AppInsightsUtils {
+  static addUserDataToRequests = (envelope: EnvelopeTelemetry, contextObjects: ContextObject): boolean => {
+    const isRequest = envelope.data.baseType === Contracts.TelemetryTypeString.Request
+    if (isRequest) {
+      const { username, activeCaseLoadId } = contextObjects?.['http.ServerRequest']?.res?.locals?.user || {}
+      if (username && envelope.data.baseData) {
+        const { properties } = envelope.data.baseData
+        // eslint-disable-next-line no-param-reassign
+        envelope.data.baseData.properties = {
+          activeCaseLoadId,
+          username,
+          ...properties,
+        }
```

Comment:
I'm guessing `activeCaseLoadId` and `username` wouldn't have been in the original `envelope.data.baseData.properties`? (So they won't be overwritten by `...properties` coming at the end rather than the start?